src/pulselistener/pulselistener/mercurial.py

```diff
@@ -53,12 +54,24 @@ def cleanup(self):
         logger.info('Removed ssh key')
 
     async def run(self):
-        # Start by updating the repo
-        logger.info('Checking out tip', repo=self.repo_url)
-        self.repo = batch_checkout(self.repo_url, self.repo_dir, batch_size=self.batch_size)
+        # Start by updating the repo in a separate process
+        clone = multiprocessing.Process(
+            target=batch_checkout,
+            args=(self.repo_url, self.repo_dir),
+            kwargs={'batch_size': self.batch_size},
+        )
+        clone.start()
+        logger.info('Checking out tip in a separate process', repo=self.repo_url, pid=clone.pid)
+        while clone.is_alive():
+            # Let the other consumers work while cloning
+            await asyncio.sleep(5)
```

Comment:
You could use [run_in_executor](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) (with a ProcessPoolExecutor with a single process) to avoid doing this hacky wait

Reply:
great tip :+1: